pythran/pythonic/numpy/expand_dims.hpp

```diff
@@ -0,0 +1,40 @@
+#ifndef PYTHONIC_NUMPY_EXPAND_DIMS_HPP
+#define PYTHONIC_NUMPY_EXPAND_DIMS_HPP
+
+#include "pythonic/include/numpy/expand_dims.hpp"
+#include "pythonic/numpy/ndarray/reshape.hpp"
+#include "pythonic/utils/functor.hpp"
+#include "pythonic/include/utils/array_helper.hpp"
+#include "pythonic/types/ndarray.hpp"
+
+PYTHONIC_NS_BEGIN
+
+namespace numpy
+{
+  template <typename T>
+  types::ndarray<typename T::dtype, types::array<long, T::value + 1>>
+  expand_dims(T const &input, int axis)
+  {
+    const int N = std::tuple_size<decltype(input._shape)>::value;
+    if (axis == -1)
+      axis += N+1;
+    std::array<long, N + 1> dim_array;
+    auto in_shape = sutils::array(input._shape);
+    long ii, jj;
+    for (ii = jj = 0; ii < N + 1; ii++) {
+      if (ii == axis) {
+        dim_array[ii] = 1;
+      } else {
+        dim_array[ii] = in_shape[jj++];
+      }
+    }
+    sutils::push_front_t<decltype(input._shape), long> new_shape;
```

Comment:
this is incorrect, as your return type specifies ` types::array<long, T::value + 1>` as return type. why not reshaping directly using `dim_array` ?

Reply:
Yes, much simpler, I didn't realize I could pass an array, instead of a pshape.

Reply:
Serge, if I try reshaping with
```
return ndarray::reshape(input, dim_array);
```
compilation fails. I'm having a hard time figuring out whether reshape can accept an array as input....